src/base/ERC721Permit_v4.sol

```diff
@@ -27,13 +27,49 @@ abstract contract ERC721Permit_v4 is ERC721, IERC721Permit_v4, EIP712_v4, Unorde
         address owner = ownerOf(tokenId);
         if (spender == owner) revert NoSelfPermit();
 
-        bytes32 hash = ERC721PermitHashLibrary.hash(spender, tokenId, nonce, deadline);
+        bytes32 hash = ERC721PermitHashLibrary.hashPermit(spender, tokenId, nonce, deadline);
         signature.verify(_hashTypedData(hash), owner);
 
         _useUnorderedNonce(owner, nonce);
         _approve(owner, spender, tokenId);
     }
 
+    /// @inheritdoc IERC721Permit_v4
+    function permitForAll(
+        address owner,
+        address operator,
+        bool approved,
+        uint256 deadline,
+        uint256 nonce,
+        bytes calldata signature
+    ) external payable {
+        if (block.timestamp > deadline) revert DeadlineExpired();
```

Comment:
shared modifier? We should also make the revert name more distinct since we have `DeadlinePassed` in modifyLiquidities